tools/amazon_agent_settings/prepare_userdata

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+
+function setup_ruby() {
+version=$1
+
+cat <<EOF
+#!/bin/bash
+
+function setup_ruby() {
+  yum -y update
+  yum -y install git-core zlib zlib-devel gcc-c++ patch readline readline-devel libyaml-devel libffi-devel openssl-devel make bzip2 autoconf automake libtool bison curl sqlite-devel postgresql-devel
+  git clone https://github.com/rbenv/rbenv.git ~/.rbenv
```

Comment:
:-1: Prefer chruby and ruby-install like we do in manageiq-appliance-build and our Dockerfile.  See https://github.com/ManageIQ/manageiq/blob/master/Dockerfile#L97-L110 for example.